### ISSUE TITLE
Segmentation fault when passing tf.string tensor arguments in TFLite

### ISSUE THREAD

**Issue by leandro-gracia-gil** [tensorflow/tensorflow#53111]
### 1. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04
- TensorFlow installation (pip package or built from source): pip
- TensorFlow library (version, if pip package or github SHA, if built from source): at least TF 2.7 and tf-nightly are affected

### 2. Code

```python
import tensorflow as tf                                                        
                                                                               
class TestModel(tf.keras.models.Model):                                        
  
  def __init__(self, **kwargs):
    super().__init__(**kwargs)                                                 
    self._hash = tf.lookup.StaticHashTable(
        tf.lookup.KeyValueTensorInitializer(
            tf.constant(['testing', 'this', 'thing']),                         
            tf.constant([1, 2, 3])),
        default_value=-1)                                                      

  @tf.function
  def test(self, word):
    return self._hash.lookup(word)                                             


test_model = TestModel()                                                       
signatures = [test_model.test.get_concrete_function(tf.TensorSpec([None], tf.string))] 
    
converter = tf.lite.TFLiteConverter.from_concrete_functions(signatures, test_model) 
converter.optimizations = [tf.lite.Optimize.DEFAULT]                           
converter.target_spec.supported_ops = [                                        
  tf.lite.OpsSet.TFLITE_BUILTINS,
  tf.lite.OpsSet.SELECT_TF_OPS                                                 
]

tflite_model = converter.convert()                                             
interpreter = tf.lite.Interpreter(model_content=tflite_model)

# Causes segmentation fault. Running test_model.test directly works fine.
result = interpreter.get_signature_runner()(word=tf.constant(['testing', 'that', 'thing']))
```

### 3. Failure after conversion
Conversion raises the following warning but completes.

```
2021-11-18 22:35:48.110926: W tensorflow/compiler/mlir/lite/flatbuffer_export.cc:1880] Graph contains the following resource op(s), that use(s) resource type. Currently, the resource type is not natively supported in TFLite. Please consider not using the resource type if there are issues with either TFLite converter or TFLite runtime:
Resource ops: HashTableV2, LookupTableFindV2, LookupTableImportV2
Details:
	tf.HashTableV2() -> (tensor<!tf_type.resource>) : {container = "", device = "", key_dtype = !tf_type.string, shared_name = "13", use_node_name_sharing = false, value_dtype = i32}
	tf.LookupTableFindV2(tensor<!tf_type.resource>, tensor<?x!tf_type.string>, tensor<i32>) -> (tensor<*xi32>) : {device = ""}
	tf.LookupTableImportV2(tensor<!tf_type.resource>, tensor<3x!tf_type.string>, tensor<3xi32>) -> () : {device = ""}
```

However, trying to run the signature fails with a segmentation fault.

I thought that static hash table lookups were supported in TFLite since many versions ago. Any possible temporary workarounds until this is fixed would be very much appreciated.

**Comment by mohantym** [comment]
Hi @sanatmpa1! Could you look into this issue. It is replicating in TF 2.5,2.6 and nightly . Attaching [Gist](https://colab.research.google.com/gist/mohantym/251be9d11a652fa6bfafe773e01e24bf/github_53111.ipynb) for Reference.

**Comment by sanatmpa1** [comment]
@leandro-gracia-gil,

I tried reproducing the issue, where I can see that session getting crashed after running the code.But I don't see the warning you mentioned above. Here's the [gist](https://colab.sandbox.google.com/gist/sanatmpa1/b2f148e8912019c70e6122de410c630b/github_53111.ipynb) and can you clarify on the same? Thanks!

**Comment by leandro-gracia-gil** [comment]
@sanatmpa1 

I'm happy to try providing more info, but I'm not sure what you need. I can confirm you that for me running the exact code I pasted above in TF 2.7 does show that warning among other outputs, and also the keras warning present in the gist. Could it be that the gist is simply less verbose?

Here's a more detailed output:  (I've omitted the initial lines including hostnames and available CUDA devices and versions)
```
WARNING:tensorflow:Skipping full serialization of Keras layer <__main__.TestModel object at 0x7f4fa1c49790>, because it is not built.
2021-11-25 21:17:45.790533: W tensorflow/python/util/util.cc:368] Sets are not currently considered sequences, but this may change in the future, so consider avoiding using them.
2021-11-25 21:17:46.010478: W tensorflow/compiler/mlir/lite/python/tf_tfl_flatbuffer_helpers.cc:363] Ignored output_format.
2021-11-25 21:17:46.010513: W tensorflow/compiler/mlir/lite/python/tf_tfl_flatbuffer_helpers.cc:366] Ignored drop_control_dependency.
2021-11-25 21:17:46.011638: I tensorflow/cc/saved_model/reader.cc:43] Reading SavedModel from: /tmp/tmp4h39x40o
2021-11-25 21:17:46.012182: I tensorflow/cc/saved_model/reader.cc:107] Reading meta graph with tags { serve }
2021-11-25 21:17:46.012213: I tensorflow/cc/saved_model/reader.cc:148] Reading SavedModel debug info (if present) from: /tmp/tmp4h39x40o
2021-11-25 21:17:46.015451: I tensorflow/cc/saved_model/loader.cc:210] Restoring SavedModel bundle.
2021-11-25 21:17:46.026026: I tensorflow/cc/saved_model/loader.cc:194] Running initialization op on SavedModel bundle at path: /tmp/tmp4h39x40o
2021-11-25 21:17:46.042800: I tensorflow/cc/saved_model/loader.cc:283] SavedModel load for tags { serve }; Status: success: OK. Took 31164 microseconds.
2021-11-25 21:17:46.053337: I tensorflow/compiler/mlir/tensorflow/utils/dump_mlir_util.cc:237] disabling MLIR crash reproducer, set env var `MLIR_CRASH_REPRODUCER_DIRECTORY` to enable.
2021-11-25 21:17:46.089494: W tensorflow/compiler/mlir/lite/flatbuffer_export.cc:1880] Graph contains the following resource op(s), that use(s) resource type. Currently, the resource type is not natively supported in TFLite. Please consider not using the resource type if there are issues with either TFLite converter or TFLite runtime:
Resource ops: HashTableV2, LookupTableFindV2, LookupTableImportV2
Details:
        tf.HashTableV2() -> (tensor<!tf_type.resource>) : {container = "", device = "", key_dtype = !tf_type.string, shared_name = "13", use_node_name_sharing = false, value_dtype = i32}
        tf.LookupTableFindV2(tensor<!tf_type.resource>, tensor<?x!tf_type.string>, tensor<i32>) -> (tensor<*xi32>) : {device = ""}
        tf.LookupTableImportV2(tensor<!tf_type.resource>, tensor<3x!tf_type.string>, tensor<3xi32>) -> () : {device = ""}
2021-11-25 21:17:46.089591: W tensorflow/compiler/mlir/lite/flatbuffer_export.cc:1891] TFLite interpreter needs to link Flex delegate in order to run the model since it contains the following Select TFop(s):
Flex ops: FlexHashTableV2, FlexLookupTableFindV2, FlexLookupTableImportV2
Details:
        tf.HashTableV2() -> (tensor<!tf_type.resource>) : {container = "", device = "", key_dtype = !tf_type.string, shared_name = "13", use_node_name_sharing = false, value_dtype = i32}
        tf.LookupTableFindV2(tensor<!tf_type.resource>, tensor<?x!tf_type.string>, tensor<i32>) -> (tensor<*xi32>) : {device = ""}
        tf.LookupTableImportV2(tensor<!tf_type.resource>, tensor<3x!tf_type.string>, tensor<3xi32>) -> () : {device = ""}
See instructions: https://www.tensorflow.org/lite/guide/ops_select
2021-11-25 21:17:46.089635: I tensorflow/compiler/mlir/lite/flatbuffer_export.cc:1962] Estimated count of arithmetic ops: 0  ops, equivalently 0  MACs

Estimated count of arithmetic ops: 0  ops, equivalently 0  MACs
INFO: Created TensorFlow Lite delegate for select TF ops.
INFO: TfLiteFlexDelegate delegate: 2 nodes delegated out of 3 nodes with 1 partitions.

Segmentation fault (core dumped)
```

**Comment by leandro-gracia-gil** [comment]
Important update: this does not seem to be specifically a StaticHashTable problem, but rather a problem of passing string tensor arguments to exported functions in TFLite!

This version of the function causes a segmentation fault just the same:
```python
  @tf.function
  def test(self, word):
    return word
```

The crash does not happen if changing the argument type in the signature from string to something else like tf.int32.

**Comment by pjpratik** [comment]
@leandro-gracia-gil I have reproduced the code in TF 2.7, TF 2.11 and TF Nightly. The session crashes in TF 2.7 but works fine in TF 2.11 and TF Nighlty. Could you please check this [gist](https://colab.sandbox.google.com/gist/pjpratik/338b2bbd353dd45f6cf08889aaa26417/53111.ipynb) and let us know if that helps. Thank you.

**Comment by leandro-gracia-gil** [comment]
Can confirm that the problem seems solved in TF 2.11. Closing the issue. Thanks!
